README.md

```diff
@@ -1,41 +1,125 @@
+[![codecov](https://codecov.io/gh/binary-com/deriv-app/branch/dev/graph/badge.svg?token=LClg2rlZ4z)](https://codecov.io/gh/binary-com/deriv-app)
 
-```
-$ yarn
-```
+**In this document**:
 
-### Local Development
+- [Other Documents](#other-documents)
+- [Pre-installation](#Pre-installation)
+- [Quick start](#Quick-start)
+- [Packages](#packages)
+- [Working With This Repo](#working-with-this-repo)
+  - [Package names](#package-names)
+  - [Usage](#usage)
+    - [Starting a Development Server](#starting-a-dev-server)
+    - [How to Clean Packages](#how-to-clean-packages)
+    - [Examples of Script Usage](#examples-of-script-usage)
+    - [Release](#release)
+- [PR Guidelines](#pr-guidelines)
+- [FAQ](#faq)
 
-```
-$ yarn start
-```
+## Other documents:
 
-This command starts a local development server and opens up a browser window. Most changes are reflected live without having to restart the server.
+- [General](docs/README.md) - Contains general philosophy and overview of this package
+- [Stylesheet guidelines](docs/Stylesheet/README.md) - Contains rules for CSS/SASS code style
+- [JavaScript guidelines](docs/JavaScript/README.md) - Contains rules for JS/JSX code style
+- [Modules docs](docs/Modules/README.md) - Contains implementation guides (i.e., scaffolding, code usage)
+- [e2e and performance testing docs](e2e_tests/README.md) - Contains documents for create and running e2e and performance tests
+- [Manage dependencies](docs/Dependencies/README.md)
 
-### Build
+[comment]: <> (TODO: Refactor Clean Project to be under usage)
 
-```
-$ yarn build
-```
+## Pre-installation
 
-This command generates static content into the `build` directory and can be served using any static contents hosting service.
+Before running or contribute to this project, you need to have the setup of the following package in your environment.
 
-### Deployment
+- node >=16.16.0
+- npm >=7.21.0
+- docusaurus >=2.0.0
+- linaria >=4.1.2
+- git (for `contribution`)
 
-Using SSH:
+**Note**: `node -v` and `sudo node -v` should be the same version.
 
-```
-$ USE_SSH=true yarn deploy
-```
+## Quick start
 
-Not using SSH:
+1.  **Fork the project**
 
-```
-$ GIT_USER=<Your GitHub username> yarn deploy
-```
+    In order to work on your own version of the Deriv application, please fork the project to your own repository.
 
-If you are using GitHub pages for hosting, this command is a convenient way to build the website and push to the `gh-pages` branch.
+2.  **Clone using SSH**
+
+    ```sh
+    git clone git@github.com:binary-com/deriv-api-docs.git
+    ```
+
+3.  **Enter project directory**
+
+    ```sh
+    cd deriv-api-docs
+    ```
+
+4.  **Install your dependencies:**
+
+    ```sh
+    npm i
+    ```
+
+5.  **Build the project:**
+
+    ```sh
+    npm run build
+    ```
+
+<br />
+### Starting a Development Server
+
+If you wish to work on Core, simply run `npm run start`.
+
+<br />
+
```

Comment:
```suggestion
And if you wish to work on cms configs, please run: 
```sh
npm run start:auth
```
in a separate terminal, this will start the netlify-cms-proxy-server and will enable you to work with the cms locally. 
and then run: 
```sh
npm run start
```
in another terminal. 
```
please add this part as well, thanks :pray: